R/hydromad.object.R

```diff
@@ -0,0 +1,369 @@
+    cat("Start = ", zoo::index2char(index(rx)[1], frequency(rx)),
+      ", End = ", zoo::index2char(index(rx)[NROW(rx)], frequency(rx)),
+      "\n",
+      sep = ""
+    )
+    cat("\n")
+    for (which in c("sma", "routing")) {
+      if (!is.null(x[[which]])) {
+        if (which == "sma") {
+          cat("SMA Parameters:\n")
+        } else {
+          cat("Routing Parameters:\n")
+        }
+        if (isFullySpecified(x, which = which)) {
+          ## all unique parameter values
+          coefx <- coef(x, which = which)
+          print(coefx, digits = digits, quote = FALSE, print.gap = 2)
+          if ((which == "routing") &&
+            isTRUE(x$routing %in% c("armax", "expuh"))) {
+            tmp <- describeTF(coefx)
+            if (!is.null(tmp) && !is.na(tmp)) {
+              cat("TF Structure:", tmp, "\n")
+            }
+          }
+        } else {
+          ## one or more parameters specified as ranges only
+          parlist <- coef(x, which = which, warn = FALSE)
+          print(data.frame(
+            lower = sapply(parlist, min),
+            upper = sapply(parlist, max),
+            ` ` = sapply(parlist, function(p) {
+              ## mark fixed parameters
+              if (isTRUE(diff(range(p)) == 0)) "(==)" else ""
+            }), check.names = FALSE
+          ),
+          digits = digits
+          )
+        }
+        # cat("\n")
+      }
+    }
+    if (!is.null(x$feasible.set)) {
+      cat("Feasible parameter set:\n")
+      print(apply(x$feasible.set, 2, function(xi) {
+        signif(c(lower = min(xi), upper = max(xi)), digits = digits)
+      }))
+    }
+    if (!is.null(x$rfit)) {
+      cat(
+        "Routing fit spec.:",
+        toString(deparse(x$rfit, control = c(), width = 500),
+          width = getOption("width")
+        ), "\n"
+      )
+    }
+    if (!is.null(x$fit.call)) {
+      cat("\nFit: ($fit.result)\n")
+      print(x$fit.call)
+      cat(
+        "    ", x$funevals, "function evaluations in",
+        x$timing[3], "seconds", "\n"
+      )
+    }
+    if (length(x$info.rfit) > 0) {
+      cat(
+        "\nRouting fit info: ",
+        toString(deparse(x$info.rfit, control = c(), width = 500),
+          width = getOption("width")
+        ), "\n"
+      )
+    }
+    if (!is.null(x$msg)) {
+      cat("\nMessage:", toString(x$msg), "\n")
+    }
+    invisible(x)
+  }
+
+
+#' @rdname hydromad.object
+#' @export
+str.hydromad.runlist <-
+  function(object, ...) {
+    cat("\nList of Hydromad model runs:\n")
+    str(lapply(object, function(obj) {
+      if (!is.null(obj$msg)) {
+        list(call = obj$call, message = obj$msg)
+      } else {
+        obj$call
+      }
+    }))
+    invisible()
+  }
+
+
+#' @rdname hydromad.object
+#' @export
+isValidModel <- function(object, ...) {
+  UseMethod("isValidModel")
+}
+
+
+#' @rdname hydromad.object
+#' @export
+isValidModel.default <- function(object, ...) {
+  return(FALSE)
+}
+
+
+#' @rdname hydromad.object
+#' @export
```

Comment:
What filename were these all defined in previously?

Reply:
Looks like these were originally from hydromad.R: [https://github.com/josephguillaume/hydromad/blob/818627db2e51b17aa4718f8b9119821764bbc4fd/R/hydromad.R](https://github.com/josephguillaume/hydromad/blob/818627db2e51b17aa4718f8b9119821764bbc4fd/R/hydromad.R). Happy to add them back.

Reply:
I think I'm ok keeping them here - it makes sense to keep functions documented together in a single file.
However, it looks like there are now duplicate definitions of these functions in both files - could you please delete them from hydromad.R?
Thanks!